packages/flutter_tools/test/general.shard/build_system/depfile_test.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/build_system/depfile.dart';
+
+import '../../src/common.dart';
+import '../../src/testbed.dart';
+
+void main() {
```

Comment:
Some negative tests here would be good.

Reply:
Added test cases for malformed file, duplicates, and whitespace